src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -298,7 +298,11 @@ public WorkspaceViewModel(WorkspaceModel model, DynamoViewModel dynamoViewModel)
             //respond to collection changes on the model by creating new view models
             //currently, view models are added for notes and nodes
             //connector view models are added during connection
-            Model.Nodes.CollectionChanged += Nodes_CollectionChanged;
+
+            Model.NodeAdded += Model_NodeAdded;
+            Model.NodeRemoved += Model_NodeRemoved;
+            Model.NodesCleared += Model_NodesCleared;
```

Comment:
I guess we never dispose of these event subscriptions?  Would be nice to add that unless it feels risky.  :(